src/mmap.rs

```diff
@@ -315,54 +320,53 @@ impl GuestMemoryMmap {
     /// Creates a container and allocates anonymous memory for guest memory regions.
     /// Valid memory regions are specified as a Vec of (Address, Size) tuples sorted by Address.
     pub fn new(ranges: &[(GuestAddress, usize)]) -> result::Result<Self, Error> {
-        if ranges.is_empty() {
-            return Err(Error::NoMemoryRegion);
-        }
-
-        let mut regions = Vec::<GuestRegionMmap>::new();
-        for range in ranges.iter() {
-            if let Some(last) = regions.last() {
-                if last
-                    .guest_base
-                    .checked_add(last.mapping.len() as GuestUsize)
-                    .map_or(true, |a| a > range.0)
-                {
-                    return Err(Error::MemoryRegionOverlap);
-                }
-            }
-
-            let mapping = MmapRegion::new(range.1).map_err(Error::MmapRegion)?;
-            regions.push(GuestRegionMmap {
-                mapping,
-                guest_base: range.0,
-            });
-        }
+        Self::with_files(ranges.iter().map(|r| (r.0, r.1, None)))
+    }
 
-        Ok(Self {
-            regions: Arc::new(regions),
-        })
+    /// Creates a container and allocates anonymous memory for guest memory regions.
+    /// Valid memory regions are specified as a Vec of (Address, Size, Option<FileOffset>)
+    /// tuples sorted by Address.
+    pub fn with_files<A, T>(ranges: T) -> result::Result<Self, Error>
```

Comment:
Looks good, I feel that long term the constructor `new()` could be removed.

Reply:
It was used in some tests and it looks more convenient when someone doesn't require file mappings. I can remove it if you are totally against it, but otherwise looks relatively useful for such a small method IDK :-s

Reply:
Fine with me, you can keep it.